### PR TITLE
Focus for address edit view and validation alert

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Fixes focus for 10182 form during address validation flow
- Inside EditContactInfo component, we are checking the state of the modal/editing UI, and if the previous state was 'addressValidation' and the new state is 'mailingAddress' we are triggering the focus onto the correct H3.
  - this prevents unneeded calls to focus for other contact info like mobile phone or email as well 
- Also fixes the focus being lost when the address validation alert shows up by using `waitForRenderThenFocus` within the `AddressValidationView`
- Removes one unneeded focus call where the alert would be focused after succesful update, and then immediately focus on the edit button. The alert is read by a screen reader when it is shown, so the first focus call was not needed, and it also introduced flakiness to the e2e tests

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/61243

## Testing done

- Tested locally
- Updating E2E tests for address validation

## Screenshots

No visible changes made, just focus adjustment

## What areas of the site does it impact?

10182 contact info focus, Profile application -> Contact Information

## Acceptance criteria

- [x] Focus is set correctly

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.

### CODE REVIEW

File: src/applications/appeals/10182/config/form.js [hunk 1]
```diff
@@ -50,6 +51,16 @@ import {
...
+// add custom focust for the contact info mailing address edit page
+const confirmContactInfoEditMailingAddress = {
+  ...contactInfo.confirmContactInfoEditMailingAddress,
+  scrollAndFocusTarget: () => {
+    // even when not having this focusElement call the h3 still gets focused, sooo not sure if I should include this
+    // or just leave it blank since it's the default behavior if scrollAndFocusTarget is set to an empty function
+    focusElement('#main h3');
+  },
+};
+
```
Comment (adamwhitlock1) [inline src/applications/appeals/10182/config/form.js:63]: @Mottie Curious about how you would want to do this. Its a quick and dirty fix right now and mostly just a working example of how I fixed the focus issue for that page.

Comment (Mottie) [reply src/applications/appeals/10182/config/form.js:63]: Setting `useCustomScrollAndFocus` in the `config/form` file shifts focus from the `H2` to the `H3`, but only on initial render. If you think it works better with this focus function, then let's include it!

**Review by dellerbie — APPROVED** [review]
besides the linting warnings, lgtm

File: src/applications/appeals/10182/config/form.js [hunk 2]
```diff
@@ -50,6 +51,17 @@ import {
 import manifest from '../manifest.json';
 import hearingType from '../pages/hearingType';
 
+/**
+ * add custom focust for the contact info mailing address edit page
```
[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply src/applications/appeals/10182/config/form.js]: fixed

File: src/applications/appeals/10182/config/form.js [hunk 3]
```diff
@@ -50,6 +51,17 @@ import {
...
 
+/**
+ * add custom focus for the contact info mailing address edit page
+ * fixes issue [#61243](https://github.com/department-of-veterans-affairs/va.gov-team/issues/61243)
+ */
+const confirmContactInfoEditMailingAddress = {
+  ...contactInfo.confirmContactInfoEditMailingAddress,
+  scrollAndFocusTarget: () => {
+    focusElement('#main h3');
+  },
```
Comment (Mottie) [inline src/applications/appeals/10182/config/form.js:62]: Would it be better to add this as another parameter within the [`ContactInfo` component](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/components/ContactInfo.jsx#L55)? The `scrollAndFocusTarget` callback is ignored until the form config `useCustomScrollAndFocus` is set to `true`

**Review by Mottie — APPROVED** [review]
Looks good. Maybe we can discuss updating the platform `ContactInfo` component, since this might end up as a more global problem? We have 3 appeals forms using this pattern. If you think it's good as-is, I can update the other 2 forms. Thanks!

File: src/applications/appeals/10182/config/form.js (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/10182/config/form.js:62]: Thank you! <3

File: src/applications/appeals/10182/config/form.js (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/10182/config/form.js]: Looks good! Thanks!

**Comment by adamwhitlock1** [comment]
@Mottie what do you think of these newest changes? I think this should fix all issues for the address validation cancelling.

**Review by tlei123 — APPROVED** [review]
LGTM

**Pull request merged**